### ISSUE TITLE
OnExistingUsersDetected bad responses

### ISSUE THREAD

**Issue by SeeingBlue** [swiftyspiffy/TwitchLib#97]
This is how I am joining channels:
```
    Public Sub client_onConnected(sender As Object, e As OnConnectedArgs) Handles client.OnConnected
        client.SendRaw("CAP REQ :twitch.tv/membership")
        client.SendRaw("CAP REQ :twitch.tv/commands")
        client.SendRaw("CAP REQ :twitch.tv/tags")
        For Each ch As ChannelTab In stabControl.TabPages
            client.JoinChannel(ch.Text)
        Next
    End Sub
```
It's been working find until today when I tried to join multiple channels at once, like 5+-. These are the results that are being returned:
![capture](https://cloud.githubusercontent.com/assets/7357172/21472213/2a888b22-ca9c-11e6-9c1b-b014bc90ac2b.PNG)

I am not sure if I'm doing something wrong, if IRC is misbehaving or what...

**Comment by swiftyspiffy** [comment]
Would you be able to paste the entire raw IRC string here so I can test it.

Also, as a side note you probably want to add a delay between joining channels, per Twitch guidelines. Perhaps 1 second delay between sending join commands.

**Comment by SeeingBlue** [comment]
This is what it looks like when joining 1 channel:
```
[7:05:57 PM][RAW]: :blubott!blubott@blubott.tmi.twitch.tv JOIN #deejayknight
[7:05:57 PM][RAW]: @badges=;color=#1E90FF;display-name=BluBott;emote-sets=0;mod=0;subscriber=0;user-type= :tmi.twitch.tv USERSTATE #deejayknight
[7:05:57 PM][RAW]: @broadcaster-lang=;emote-only=0;followers-only=-1;r9k=0;slow=0;subs-only=0 :tmi.twitch.tv ROOMSTATE #deejayknight
[7:05:57 PM][RAW]: :blubott.tmi.twitch.tv 353 blubott = #deejayknight :scottflorence lazloelonghair orecalgaming caerbannograbbit deejayknight zowsimon shadowninga3 epiclalagirl doofyst rafalo15 kilo_spartan deadnedz007 thundergod97 auricuoro slywolf708 wgrates queen_bad saltyoldoneeyedcondor skartung joseker96 maxedoutgamers pashc beerspecialist gnarkky xanatax brandons101 bloodreaperwolf danieleisler qwii_ sdeligar stottin gomez911 thewinterwolf grizz_warmace cabledx48 libralynx
[7:05:57 PM][RAW]: :blubott.tmi.twitch.tv 353 blubott = #deejayknight :tempocrash streamproio zeppy888 fatdox8 harmceo muxybot mainbone rastork superswinez neorxmd keetler whoatemyeggo the_amazing_schneiderman fatd33r esp4him ilbrec talung phynicks2 laskergo orbitjuice flylogger soccerpj9 kreativmind9 brmouzone senorcristos blackboxking chronowise stuccaro j11963 pilinga51 icarusgroup knightcorpsbot kaoiji thegoatmilkguy gerald_lopez certauthentic scipio_germanicus slagmetal rottennail
[7:05:57 PM][RAW]: :blubott.tmi.twitch.tv 353 blubott = #deejayknight :vandalg loki_loco paullybw canisofwar wawaweeewa mcicy tsundereadmiral toiletpoper kingnewbs t0ghar branebot rcil arisboch chygrynshky galino2606 time_shift rorybellows proxus571 mech82 quiet50ul ipawk war77el millertime610 curseappbot yueizen stanstanman lumarious
[7:05:57 PM][RAW]: :blubott.tmi.twitch.tv 353 blubott = #deejayknight :blubott
[7:05:57 PM][RAW]: :blubott.tmi.twitch.tv 366 blubott #deejayknight :End of /NAMES list
```

This is what joining 2 channels looks like:
```
[7:02:24 PM][RAW]: :blubott!blubott@blubott.tmi.twitch.tv JOIN #deejayknight
[7:02:24 PM][RAW]: @badges=;color=#1E90FF;display-name=BluBott;emote-sets=0;mod=0;subscriber=0;user-type= :tmi.twitch.tv USERSTATE #deejayknight
[7:02:24 PM][RAW]: @broadcaster-lang=;emote-only=0;followers-only=-1;r9k=0;slow=0;subs-only=0 :tmi.twitch.tv ROOMSTATE #deejayknight
[7:02:24 PM][RAW]: :blubott.tmi.twitch.tv 353 blubott = #deejayknight :esp4him mech82 bloodreaperwolf the_amazing_schneiderman phynicks2 streamproio talung knightcorpsbot quiet50ul epiclalagirl beerspecialist stuccaro ilbrec rcil orecalgaming deadnedz007 flylogger curseappbot branebot sdeligar kreativmind9 slagmetal loki_loco vandalg lumarious chronowise blackboxking certauthentic thundergod97 canisofwar shadowninga3 saltyoldoneeyedcondor toiletpoper auricuoro rastork chygrynshky queen_bad
[7:02:24 PM][RAW]: :blubott.tmi.twitch.tv 353 blubott = #deejayknight :doofyst scottflorence skartung kilo_spartan gomez911 senorcristos zeppy888 xanatax maxedoutgamers icarusgroup arisboch stanstanman laskergo wgrates gnarkky zowsimon ipawk qwii_ fatd33r brmouzone whoatemyeggo wawaweeewa fatdox8 scipio_germanicus war77el t0ghar j11963 muxybot rorybellows caerbannograbbit pilinga51 galino2606 slywolf708 grizz_warmace proxus571 keetler millertime610 rottennail orbitjuice lazloelonghair
[7:02:24 PM][RAW]: :blubott.tmi.twitch.tv 353 blubott = #deejayknight :cabledx48 paullybw rafalo15 thewinterwolf joseker96 kingnewbs danieleisler soccerpj9 neorxmd mainbone deejayknight mcicy brandons101 harmceo stottin time_shift gerald_lopez superswinez kaoiji tsundereadmiral pashc tempocrash libralynx thegoatmilkguy yueizen
[7:02:24 PM][RAW]: :blubott.tmi.twitch.tv 353 blubott = #deejayknight :blubott
[7:02:24 PM][RAW]: :blubott.tmi.twitch.tv 366 blubott #deejayknight :End of /NAMES list
[7:02:24 PM][RAW]: :jtv MODE #deejayknight +o war77el
[7:02:25 PM][RAW]: :jtv MODE #deejayknight +o deejayknight
[7:02:25 PM][RAW]: :jtv MODE #deejayknight +o knightcorpsbot
[7:02:25 PM][RAW]: :jtv MODE #deejayknight +o epiclalagirl
[7:02:25 PM][RAW]: :jtv MODE #deejayknight +o branebot
**************Placing Break Here**************
[7:02:25 PM][RAW]: :blubott!blubott@blubott.tmi.twitch.tv JOIN #monstercat
[7:02:25 PM][RAW]: @badges=;color=#1E90FF;display-name=BluBott;emote-sets=0;mod=0;subscriber=0;user-type= :tmi.twitch.tv USERSTATE #monstercat
[7:02:25 PM][RAW]: @broadcaster-lang=;emote-only=0;followers-only=-1;r9k=0;slow=0;subs-only=0 :tmi.twitch.tv ROOMSTATE #monstercat
[7:02:25 PM][RAW]: :blubott.tmi.twitch.tv 353 blubott = #monstercat :ste97casty funeila gianluk9gianluca h20wyvern kanehoho kittylover2605 gnivrihog ksixtapoutx lazarus_octern zaneoblaneo imtwitchinbruh alpinetree timberwolf920 pakunsio darkmit0 beetrootuniverse inextafro oblivion__official givmig bjohei03 tweekazbns salicsa bydreamy99 eduardo272727 applescdr teeleo bucolucas littletinyo nocturnal_rover rytnek jamesdoesgamesxyt faxfox john_wally marasias thesquishyrs octarockstudio1
[7:02:25 PM][RAW]: :blubott.tmi.twitch.tv 353 blubott = #monstercat :lyralulamoon abkayckay dkmintz alphakappa099 matt1414141 johnny7c3 nojiko12 notoriousthief agentsfire lapron3000 sharagran iishyguyii merk890 saynt25 thedarkwaypoint borvion ante_taichou bell_cs da_wiggler fer112233mc 10xephos01 yaphee cap_macmillan mainfatality crazymustard jalmaan jurszyk1337 axothic prettysalty tylerblazeit salvage_ thomel stardragon21 cazamiii iamalovelypenguin patrickman88 necrowish strohhutdave
[7:02:25 PM][RAW]: :blubott.tmi.twitch.tv 353 blubott = #monstercat :pkplays_ midnightslight silverklick jonissiasdf bobmfjones wuder4 harmon758 serioskiller brokentesla adurpadurp harrytripper prosaint82 officialdeadpool4000 martinwonder tsun13 subakii savethescreen silvermasterhd vassaraw96 ehzin knightsalot breezieboy grizzlybeard45 kiritochwan clasic_dayz1 koolguy_games lethalnightmarex3 wakkenation toshihiro_ emplodedpear techgod52 garaxe swainburg johsun81 solidseagull inet_official
[7:02:25 PM][RAW]: :blubott.tmi.twitch.tv 353 blubott = #monstercat :pizza231 antoxbl4 jasonharty elegantduck vladi2727 ragetonight fluctuantflatulence broolstoryco17 gener_r jadimini gizmodudemeow sabbrewolf grannox thisaie hopefulveteran timsfavoriteogsub jmcj07 bearhunter13 samantha9001 saveairhead summon30 mckvig thenukepenguin gonehaywyre storecasper1 sweetazu midjones gockel1998 larsreili shield_smash crioto icedliveordie 0_marslp_0 ak65234 pricyspark charazu airdap phombie
[7:02:25 PM][RAW]: :blubott.tmi.twitch.tv 353 blubott = #monstercat :geileschnidde_1337 alzaher91 elcactus21 mynameiszyper thatsabadkitty pow_pow09 darkgamer120 spartanyko repisz wallefails pb_nerd radical_vlad orels1 izhzhun tomlube xanatos67 12344556788 firewheeling myrkwell dinguschief mjnielsen99 phi321 totalschaden997 legandlagoon ttcmario elimasters neapol94 theembster kiinetix molokax aimyss majonachote zgtfo cyp21a skarnek justremmy hupplings explodingbones lordshieco enderhorse
[7:02:25 PM][RAW]: :blubott.tmi.twitch.tv 353 blubott = #monstercat :joannou1 kutlus91 callmepickle obayleyx emeth806 cegglestn goodgy secobomb oudamnnoscouped wooden_sword3 hoplite570 skeletalwaffle caprolactam6ers gaidenkz aglmegadeath nikon662 theomegaz8 gcatz_mb hell5torm ira0311 gille90 xxredxbulletxx brgrs lolplayer74 drak0v elkian_ etalim torkfat aleksveira janzozman hokebjbj gamingnerdd xsabix ela17 sephirum wolfram3 qqarioh gillymomo wallantia mcteameagle potatoeskillsyou
[7:02:25 PM][RAW]: :blubott.tmi.twitch.tv 353 blubott = #monstercat :uncreatedben bennybarygoodle theghostsfire hype348 espeneiks nijigasume mrbrinks2 infernox7 tigrejor fusrodahfgt mysticgemstone markverbeek satamapiiri swoll9 sweetiecandylove saldon_semit starlightaria das_terminator flappyflannel kingsilentt l334_m0n jaydipm melkiy582 iigaticaii wolfaxsi jabberwoky69 lorddaenerius aeromancia huo556 c0betchik theawesomelenny dbc__ millerrrtv macduffinrawr khanngy hate_4_u rafalo15
[7:02:25 PM][RAW]: :blubott.tmi.twitch.tv 353 blubott = #monstercat :xemo1234 denisclau01 shinjo2040 playingmankee noepel spaanplaat zandafied jamaeyy mewnlightrose some1another dekrass jxtwitchx murthias neshara og_rhyze oate1 superstreamstrike queromais3 ptinoz stscarhand hippohilde doopey655 instinct_fury acnik cpttylor20 phoenixx90 ko_li_es blaksadow123 xxdatfrodoexx olironmael cadillaxx useamane exploitation5 grouzik193 oscarxcore docteck pandentia hamator2302 blackpsycho55
[7:02:25 PM][RAW]: :blubott.tmi.twitch.tv 353 blubott = #monstercat :notoriousrose bones_1337 daluxmusic etorac batman1048 yanarossy logiafruit slick__77 ansidinc miss_emmy botakio kamaper 1227crazykid hexagonzzz random_49 jesskalamity pedromig123 jsl7 khofez tails_of_nine p90official nox1504 scorpionsadventure dr4gos crazykidcj heraxxje walmanp scxthe rotendo x_eldingar_x dbrandt01 toombs26 lefurtif126 sanguezap tabrise3 dergeileharalt liquidfuel cro_tera runsh skrimmy little__7 dipndawtz
[7:02:26 PM][RAW]: :blubott.tmi.twitch.tv 353 blubott = #monstercat :atolifuzen akusoru ranger730 thingsgaming alexandrobullos ryukurai bzgex endoth3lium stunja23 droboviachek bomdshell1234 kepkanumberone kongo401 gaxleg foxmjhoward dan_nuggets captainsheeper ganicara slythstream carrotpc rentvz hichem53 poookerz masterssj niftychunks maw3b doberman4ik achmedior pixfamily tommy5346 takumi_qhn curetix yipeeekaiyay rgravity xprobinx vitalblack nehel666 wayne8325 st331blade lukesrb tukbochko
[7:02:26 PM][RAW]: :blubott.tmi.twitch.tv 353 blubott = #monstercat :worldrecordbreaker dutchdj28 okaria vlad333000 litrecc cimetulja polecat youaskedforit360 entreavaux brandon4542 pinsir_v2 zaion_lol whidbeykid123 darkmagicpower hansolosc2 zarch0 theepicxblaze328 worker77777 chiktikkaquick hereticzero walnutplays whtedynamite bootsdaman ufgkirk kevindo1119 zegaria 86421357 tiagoakatheskillers monstercat rekrutenlp xlphysiczs marcusvey lillbyxan roughshod xxmcmaxx lilalotus_ fadenxc86
[7:02:26 PM][RAW]: :blubott.tmi.twitch.tv 353 blubott = #monstercat :ethancentaurai raginryan majesticredstone tiuqu belioz grlmreaver thepeacefulhorror flame223 mysticmurv deucez2710 smokedog98 justone895 shikiyoku5 ch4sing comex1234 jacky1356400 blasman13 dragonslayer205 dylcraft niebezpieczny888 fralla8 wouterr_ magesticethan49 myriadth kyleatwar thr3343 themk4u hoboman12u madaxel cofor owlbrownie becauseofme drbleifuss don_tomek varetal acegunro mlgret4rd rigleif thedoomsblade arran999
[7:02:26 PM][RAW]: :blubott.tmi.twitch.tv 353 blubott = #monstercat :flowbite pepsimaxgaming rexthebozz pliant_couger embassador_bacon peterobertwut phantom jactomhd rifted1 thuglife69878 s1r_t0b3y burningjoker madebysid kera_neew cractice06 loganwt1 360snajper miniminerlps thepencilsharpener d20dad titanium_twister8675 zocktross mysticknightgaming halvyyy jab1232 onodonut quazar117 yrlish thearatus jenovahale evyvendetta oathclaw noxilar lukasmc14 todous puetz jamaicainternational arrowzs
[7:02:26 PM][RAW]: :blubott.tmi.twitch.tv 353 blubott = #monstercat :maxpro778 revlobot arandomperson41 drothethedestroyer 170734 doomedclock dekaoftaurus sayrn pwlnism crimsongrray zerofever nightbot polkan1221 overrustlelogs helpower2 internetexplorervevo shadowminator cheereaque suspiciouston scroowd sashaday drwolf_god dominik008 fr_suomalainen reesboy89 thekillerbobz i_lost_my_account thombern bossmanxxx rawkt_csgo fwa_ blackblood971224 hiromisuto tiomun1 smoat spirit228 decicus
[7:02:26 PM][RAW]: :blubott.tmi.twitch.tv 353 blubott = #monstercat :crazzybooi ethreck wonderfultrue aaron565 hoc_havok annihalator666 aceials beanicefay iiqskii protossmudkipturtle tsevrah iamironjudy thenobleseraph freakraven slimetom space_boots ruben001100 cokimonstr17 intics_ santo123955 luxaetern s4rin epicschoggy psycho_67 gutter3052 atomsk_blade nxxxm disk0rd42 xeodk ssad0 mranonymois vash66 infinatioustv tugo28 cptirgendeiner bigarms21 gunmetal01 peevi preeskie highlywalk2
[7:02:26 PM][RAW]: :blubott.tmi.twitch.tv 353 blubott = #monstercat :stolcraft plasmafyre cmdr_spacefap pro8byfake duenkel seif100 eruuupt logviewer itsmralex levmaker almost420 redbeard_bravo wolfyportal slymartins maxreaper117 andy1686 sirmooshroom breakerspl pixelc4t sir_vatson creeperkillertm rawramatiger terdytheterd logicslayer gatorman00 t1mb3rwot alfredoryan szl_blakery mr_mstry schokokrispy tilmoe netherzoon icyman0330 kurrukz kingwindsor quanoxx gardeneur emosewa76 soulcrusher101
[7:02:26 PM][RAW]: :blubott.tmi.twitch.tv 353 blubott = #monstercat :rpescape wizytv trollblox gianttreelp korgelftw brightlurker stabmenow nodinawe thelordviper lowheartrate patrick_ku panthos b4ilout zephxyz imodzcodz venturetk spigler kuroitaiyo dacowii sharpadin durable95 sunforest ixale captsarge7 unstoppablechaos workitune ff778 understandingmadness andyadd jyxi k231i d1m0a1n miksu991 tsurara houlfala zugashii xdrpunkx sazukisenpai vladimirzalupkin stezhd rolyjulioli imfromgerman
[7:02:26 PM][RAW]: :blubott.tmi.twitch.tv 353 blubott = #monstercat :stoovs cjwprostar hotmrsoup falxxxxxxxx motanu878 skullvender thebowbros kurcide snacho felps chocolatewraith urahgan keybladexpain tumultuoustiger michaelrrox shidongan therealkleinba minysmoral nichev filet24 savemysoulfx soluxian zarbonhd jaddagames loganthebeastfu szita999 noahisunbeatable delltor123 quamquam1 f2e4k wtmike24 hazamog sorog_s nomastersnorulers thegreatdangus bluzziegaming contractspec9 curseappbot
[7:02:26 PM][RAW]: :blubott.tmi.twitch.tv 353 blubott = #monstercat :tminnich ahiiru inexist3nce frostydrinkz onslaught meujeudom ondtyrann vkanji cheese_blue catacaramel tedwyn danischu seebeback sp00nerism cory1351 as_65 prorookiemc feralio wrath220 lelle_03 pandapalooz winstonpus frostyteno lauaaflynn tomato_oshi wizhang9 supermoo sobol6803 thedavidofdoom bbalidin nutrilicious terasinube lukemoorhous seppiiplayswow macbaxter mirrass z123zocker everyonehatesalienboy prideinmyevo
[7:02:27 PM][RAW]: :blubott.tmi.twitch.tv 353 blubott = #monstercat :thedarkshepherd mehanpro jetox212 orbitperson superdave334 augrewa ghioxo crazypeanut03 php1r serpiboy firstchapter ffs____ flufkin007 legopigguy toxyy brotherbrad711 mcflufff_ terrimoto kerzyte notedbinkie9727 woddeh evol331 murder4hyre sparadin matt4756roblox odsox swiftynuggets zaeps15 minori1475369 djandywtv paulsibergod dozsa0 ghost295 cirdecxx nichb admiral_zupa thehijacker7 evila1ive hulabulia nodeadnolife1657
[7:02:27 PM][RAW]: :blubott.tmi.twitch.tv 353 blubott = #monstercat :zimtok5 warre12 ol2azerrr 99damagebot devastater323 adammcdowell overlord_kostyanik jaqquetv mrstrongoak gtx1080quadsli etienne951 sn1pone xxjok59xx zanze2 bronzeleesin420 patrossgaming ph_cool_10 bomj_kolyadun deco256 kryptonikcousin lomglpower schmodo exclusiveg13 pixxelp mrfunkyjar m1cr0man doctormoose34 bbq_can2 lordcapivara plasmagladiator hqsm mrnice9292 vertigolight nuriksakura svenyannick rainman_g qdaza redlightss
[7:02:27 PM][RAW]: :blubott.tmi.twitch.tv 353 blubott = #monstercat :die_88 val2t1 dr_scrubbington rems7393 hexyeh phantom_range demon5930 p0lt10n aviatornicboy28 thefatwizard electronization nainsc2 croqodil sirtigerthe1st chaosblackdoom g3no__ achaser04 battle3__ blade1156 nanocatz r_slattery120 caseyp89 qruralp w4bb enemy0405 novike kylpet1 pistolpete986 blakesleed doompick89 nortok_ sparatick swynnnn reepje1233 alexchio23 trigger117 fransbabyseal cmy03290 epic_yolo_swag loony555
[7:02:27 PM][RAW]: :blubott.tmi.twitch.tv 353 blubott = #monstercat :kattennisse hainesftw the_isuall
[7:02:27 PM][RAW]: :blubott.tmi.twitch.tv 353 blubott = #monstercat :blubott
[7:02:27 PM][RAW]: :blubott.tmi.twitch.tv 366 blubott #monstercat :End of /NAMES list
```

I haven't ruled out that it's something I may be doing, just not sure what it is that could be causing the OnExistingUsersDetectedArgs to do that.

**Comment by swiftyspiffy** [comment]
And those lines are what the library failed to parse, I assume? I tried joining DeejayKnight and then Monstercat and it appears to be parsing correctly.

**Comment by SeeingBlue** [comment]
Yea, I've been doing some testing. I am using a 5000ms sleep between joins. It seems like the trouble occurs when I send a second join request before the first channel's 353 comes back.
If the deejayknight's 353 comes back first then it's fine, but if 5 seconds expire and I send join request for monstercat before deejayknight's 353 then all hell breaks lose.
The first response I get back is from monstercat channel, but it contains users as deejayknights 353 header and users of his channel. The second 2 responses just contain more headers from the deejayknight 353. The fine response contains members of the monstercat channel, It always parses the second channels 353 correctly, with the exception of containing first channels junk/users.

TLDR
I never get a proper parse of the first channels 353 if I join a second channel before the first channels 353 returns.

**Comment by swiftyspiffy** [comment]
Sounds like the library would benefit from having a queuing system that joins channels at it's own rate to ensure everything works correctly. What are your thoughts on this idea? Basically you could fire JoinChannel as much as you want. The library will queue up those channels and join them sequentially after each has 100% connected and ready to go.

**Comment by SeeingBlue** [comment]
I was thinking the same thing. That required delay could vary based on load/bandwidth, so it's probably best to make sure the previous channels 366 has came before joining a new channel. It would solve this issue. Were you able to recreate it?

**Comment by swiftyspiffy** [comment]
I did not try to recreate it, but I am aware of the I guess sort of race conditions that are present here. The library doesn't account for joining multiple channels concurrently.  The 366 is a great identifier for when the existing users ends. I will implement it now. Might be a few minutes.

**Comment by SeeingBlue** [comment]
While we're on the subject. If I continue to use this library in a multi-channel fashion, do you foresee other similar issues?

**Comment by swiftyspiffy** [comment]
The library originated out of code I was using for a single channel bot. I packaged it up and posted the repo here. Shortly after I added the ability to join channels. While I think most of the library should be good, all of my test environments are single channel use only. That being said, there are a number of users who use the library in multi-channel environments. This was an interesting use case as I don't think any of them fire off join commands really quickly.

**Comment by SeeingBlue** [comment]
My bot saves the channels you were in and rejoins them after a reconnect. I'm curious how the other multichannel users would handle joining multiple channels during a reconnect. Maybe their own queuing?

**Comment by swiftyspiffy** [comment]
Right now there aren't mechanisms to reconnect and join each individual channel the bot was in.  The idea actually hasn't crossed my mind as the only issues I've had related to my own bot is runtime errors and the library can handle them fairly well. It is now on the "roadmap" though.

**Comment by SeeingBlue** [comment]
It seems to be working great now. I removed my delay, channels loaded perfect. I'll continue stressing this feature though and keep you updated if I find anything.

**Comment by swiftyspiffy** [comment]
Awesome, I'll close this for now, feel free to reopen it if you find any further issues and I can look into them!

**Comment by SeeingBlue** [comment]
I'm having intermittent issues with joining. I haven't figure out how to recreate it yet because sometimes it seems to work fine.
I first noticed it when I couldn't join more than 7 channels. I disconnected and reconnected. It started working.
Worked fine for awhile then again today I connected to 8 channels initially but now any channels I try joining afterwards just don't do anything. I'm sending the request correctly, but no response from IRC or anything. It just ignores me and continues processing existing channels like normal.
Restarted the bot, now it connects to all 9 channels and is accepting additional channels too.
Any ideas?

**Comment by swiftyspiffy** [comment]
Is it possible some of the time based limitations stated here:
https://help.twitch.tv/customer/portal/articles/1302780-twitch-irc

Are affecting it? Specifically the time based join limitations.

I'll look into it in a few hours.

**Comment by SeeingBlue** [comment]
Yes. I checked on that too. The limit is 50 joins in 15 seconds but we're talking about less than 10 here. Disconnect and reconnect then it starts working. Kinda strange but then again i thought the first issue was too.

**Comment by swiftyspiffy** [comment]
I was able to join 8 channels on top of the original channel I joined (total of 9) one after each other with a second delay in between. Will test without delay now.

**Comment by SeeingBlue** [comment]
I suspected it would work for you as it works for me most of the time. I'm still playing around with it to see if I can reproduce the issue at will. Currently it's just hit or miss then like I said a simple reconnect corrects the issue. I will keep using this and hopefully find something concrete.

**Comment by swiftyspiffy** [comment]
Any updates on this by chance?

**Comment by SeeingBlue** [comment]
Well I had an IRC library I modified for twitch that I have used on previous projects. It wasn't very difficult to swap them out. Everything has been working perfectly so while I don't have any more information as to the cause, I am more inclined to believe it's in the library as time goes on.

**Comment by swiftyspiffy** [comment]
Alright, well, I'll go ahead and close this for now. If I encounter it again, or you happen to and there's more information about it, it would be good to re-open.

**Comment by SeeingBlue** [comment]
Just curious, was any related code ever change thing might have resolved this issue?

**Comment by swiftyspiffy** [comment]
No new code has changed the way the library connects and joins channels, as far as I can recall. Also haven't had any issues come in related to connecting or joining channels since this issue.
